File: ZooJava/src/test/java/com/example/zoojava/repositories/animals/AnimalsRepositoryImplTest.java
package com.example.zoojava.repositories.animals;

import com.example.zoojava.models.Animal;
import com.example.zoojava.models.Employee;
import com.example.zoojava.models.enums.typeAnimal;
import com.example.zoojava.repositories.employees.EmployeesRepository;
import com.example.zoojava.repositories.employees.EmployeesRepositoryImpl;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.sql.SQLException;
import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Test repositorio de animales.
 * Las id de los animales son llevadas por la base de datos por lo que no sabremos su id hasta añadir al animal.
 */
class AnimalsRepositoryImplTest {

    private final AnimalsRepository repository = new AnimalsRepositoryImpl();
    private final Animal test = new Animal(1,"Kuno", typeAnimal.MAMIFEROS,LocalDate.now());


    @AfterEach
    void tearDown() {
        try {
            repository.removeAll();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
    @BeforeEach
    void setUp(){
        try {
            repository.removeAll();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    void add() throws SQLException {
        var add = repository.add(test);
        assertAll(
                () -> assertEquals(add.getName(),test.getName()),
                () -> assertEquals(add.getType(),test.getType()),
                () -> assertEquals(add.getBirthDate(),test.getBirthDate())
        );

    }

    @Test
    void remove() throws SQLException {
        var add = repository.add(test);
        var remove = repository.remove(add);
        assertAll(
                () -> assertEquals(remove.getId(),add.getId()),
                () -> assertEquals(remove.getName(),add.getName()),
                () -> assertEquals(remove.getType(),add.getType()),
                () -> assertEquals(remove.getBirthDate(),add.getBirthDate())
        );
    }

    @Test
    void findAll() throws SQLException {
        var add = repository.add(test);
        var lista = repository.findAll();
        assertAll(
                () -> assertEquals(lista.size(),1),
                () -> assertEquals(lista.get(0).getId(),add.getId()),
                () -> assertEquals(lista.get(0).getName(),add.getName()),
                () -> assertEquals(lista.get(0).getType(),add.getType()),
                () -> assertEquals(lista.get(0).getBirthDate(),add.getBirthDate())
        );
    }

    @Test
    void findById() throws SQLException {
        var add = repository.add(test);
        var find = repository.findById(add.getId());
        assertAll(
                () -> assertEquals(find.getId(),add.getId()),
                () -> assertEquals(find.getName(),add.getName()),
                () -> assertEquals(find.getType(),add.getType()),
                () -> assertEquals(find.getBirthDate(),add.getBirthDate())
        );
    }

    @Test
    void modifyById() throws SQLException {
        var add = repository.add(test);
        add.setName("cambiado");
        var modify = repository.modifyById(add.getId(),add);
        assertAll(
                () -> assertEquals(modify.getId(),add.getId()),
                () -> assertEquals(modify.getName(),add.getName()),
                () -> assertEquals(modify.getType(),add.getType()),
                () -> assertEquals(modify.getBirthDate(),add.getBirthDate())
        );
    }
}

File: ZooKotlin/src/test/kotlin/com/example/zookotlin/repositories/animals/AnimalsRepositoryImplTest.kt
package com.example.zookotlin.repositories.animals

import com.example.zookotlin.models.Animal
import com.example.zookotlin.models.enums.typeAnimal
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import java.time.LocalDate

internal class AnimalsRepositoryImplTest {

    private val repository: AnimalsRepository? = AnimalsRepositoryImpl.getInstance()
    private val test: Animal = Animal(1, "Kuno", typeAnimal.MAMIFEROS, LocalDate.now(), null)


    @AfterEach
    fun tearDown() {
        repository!!.removeAll()
    }

    @BeforeEach
    fun setUp() {
        repository!!.removeAll()
    }

    @Test
    fun add() {
        val add = repository!!.add(test)
        assertAll(
            Executable { assertEquals(add!!.getName(), test.getName()) },
            Executable { assertEquals(add!!.getType(), test.getType()) },
            Executable { assertEquals(add!!.getBirthDate(), test.getBirthDate()) },
            Executable { assertEquals(add!!.getImg(), test.getImg()) }
        )
    }

    @Test
    fun remove() {
        val add= repository!!.add(test)
        val remove = repository.remove(add!!)
        assertAll(
            Executable { assertEquals(remove!!.getId(), add.getId()) },
            Executable { assertEquals(remove!!.getName(), add.getName()) },
            Executable { assertEquals(remove!!.getType(), add.getType()) },
            Executable { assertEquals(remove!!.getBirthDate(), add.getBirthDate()) },
            Executable { assertEquals(remove!!.getImg(), add.getImg()) }
        )
    }

    @Test
    fun findAll() {
        val add = repository!!.add(test)
        val lista = repository.findAll()
        assertAll(
            Executable { assertEquals(1, lista.size) },
            Executable { assertEquals(lista[0].getId(), add!!.getId()) },
            Executable { assertEquals(lista[0].getName(), add!!.getName()) },
            Executable { assertEquals(lista[0].getType(), add!!.getType()) },
            Executable { assertEquals(lista[0].getBirthDate(), add!!.getBirthDate()) },
            Executable { assertEquals(lista[0].getImg(), add!!.getImg()) }
        )
    }

    @Test
    fun findById() {
        val add  = repository!!.add(test)
        val find = repository.findById(add!!.getId())
        assertAll(
            Executable { assertEquals(find.getId(), add.getId()) },
            Executable { assertEquals(find.getName(), add.getName()) },
            Executable { assertEquals(find.getType(), add.getType()) },
            Executable { assertEquals(find.getBirthDate(), add.getBirthDate()) },
            Executable { assertEquals(find.getImg(), add.getImg()) }
        )
    }

    @Test
    fun modifyById() {
        val add = repository!!.add(test)
        add!!.setName("cambiado")
        val modify = repository.modifyById(add.getId(), add)
        assertAll(
            Executable { assertEquals(modify!!.getId(), add.getId()) },
            Executable { assertEquals(modify!!.getName(), add.getName()) },
            Executable { assertEquals(modify!!.getType(), add.getType()) },
            Executable { assertEquals(modify!!.getBirthDate(), add.getBirthDate()) },
            Executable { assertEquals(modify!!.getImg(), add.getImg()) }
        )
    }
}